spring-batch-core/src/main/java/org/springframework/batch/core/step/tasklet/StoppableTasklet.java

```diff
@@ -38,4 +39,14 @@ public interface StoppableTasklet extends Tasklet {
 	 */
 	void stop();
 
+	/**
+	 * Used to signal that the job should stop, providing access to the current
+	 * {@link StepExecution} context.
+	 *
+	 * @param stepExecution the current {@link StepExecution} context in which the job
+	 * is being executed
+	 */
+	default void stop(StepExecution stepExecution) {
+		stop();
+	}
```

Comment:
The new method provides a **default implementation** that delegates the call to the existing stop() method, ensuring backward compatibility for implementations that do not need step-specific stopping logic.